utils/wikize_refs.py

```diff
@@ -114,274 +151,443 @@ def parse_args():
+            fns2 = re.findall("<sup>\[([a-zA-Z0-9_-]*)\],\[([a-zA-Z0-9_-]*)\]</sup>", line)
+            for fn in fns2:
+                line = re.sub("<sup>\[%s\],\[%s\]</sup>"%(fn[0],fn[1]),
+                             "<pus>[%d],[%d]</pus>"%\
+                             (ref_map[fn[0]][3]+renumber, ref_map[fn[1]][3]+renumber), line)
+            fns3 = re.findall("<sup>\[([a-zA-Z0-9_-]*)\],\[([a-zA-Z0-9_-]*)\],\[([a-zA-Z0-9_-]*)\]</sup>", line)
+            for fn in fns3:
+                line = re.sub("<sup>\[%s\],\[%s\],\[%s\]</sup>"%(fn[0],fn[1],fn[2]),
+                             "<pus>[%d],[%d],[%d]</pus>"%\
+                             (ref_map[fn[0]][3]+renumber,ref_map[fn[1]][3]+renumber,ref_map[fn[2]][3]+renumber), line)
+            line = re.sub("<pus>","<sup>", line)
+            line = re.sub("</pus>","</sup>", line)
+            outlines += [line]
+
+        else: # just output the same line as input
+            outlines += [line]
+
+    if gather_linkdefs:
+
+        for k in sorted(file_lines):
+
+            lineinfo = file_lines[k]
+            ltype = lineinfo['type']
+            line = lineinfo['line']
+
+            if ltype != 'linkdef':
+                continue
+
+            ref_hdl, ref_url, ref_tit, ref_bib = lineinfo['linkdef']
+            if magic() in line:
+                # Its already mangled from a previous run
+                outlines.append(line)
+            else:
+                # Mangle the linkdef to move it out of the way but nonetheless retain it
+                outlines.append(line.replace(ref_hdl, "%s-%s"%(ref_hdl, magic()), 1))
 
     return outlines
 
-def build_intermediate_link_defn_lines(remapped_ref_map):
+def build_intermediate_link_defn_lines(remapped_ref_map, renumber):
     """Build (auto-gen'd) intermediate link definitions"""
     outlines = []
 
     if remapped_ref_map:
-        outlines.append("\n<!--- INTERMEDIATE LINK DEFS POINT TO ANCHORS IN TABLE --->\n")
         for k,v in sorted(remapped_ref_map.items()):
-            outlines.append("[%d]: #ref%d%s\n"%(k, k, " \"%s\""%v[1] if v[1] else ""))
+            if renumber:
+                outlines.append("[%d]: #%s-%d %s\n"%(k+renumber, magic(), k+renumber, "\"%s\""%v[1] if v[1] else ""))
+            else:
+                outlines.append("[%s]: #%s-%s %s\n"%(v[3], magic(), v[3], "\"%s\""%v[1] if v[1] else ""))
 
     return outlines
 
-def build_reference_table_lines(remapped_ref_map):
-    """Build (auto-gen'd) rendered table of references"""
+def build_reference_list_lines(remapped_ref_map, renumber):
+    """Build (auto-gen'd) rendered list of references"""
     outlines = []
 
     if remapped_ref_map:
-        outlines.append("\n<!--- TABLE OF REFS RENDERED AS MARKDOWN --->\n")
-        outlines.append("References | &nbsp;\n")
-        outlines.append(":--- | :---\n")
-        for k,v in sorted(remapped_ref_map.items()):
+        outlines.append("### References <!-- (%s) -->\n"%magic())
```

Comment:
This lines messes up the formatting in on the preview.bssw.io site.  For example, if you look at the converted file ??? at:

* https://preview.bssw.io/blog_posts/celebrating-apollo-s-50th-anniversary-users-stories-from-space

it shows `References <!-- (sfer-ezikiw) -->` in the HTML output:

![image](https://user-images.githubusercontent.com/1959736/125690585-6c803fb8-ee85-46e9-bd64-3bcf12053f6d.png)

Therefore, I don't think we can merge this PR as is as it would show

Reply:
Yes, this is the key issue and it is yet another odd behavior that works fine with *normal* markdown but does not work on the backend site. We need to chat with @bernhold about this and about whatever flags/triggers we can stuff into the *references* stuff at the end to enable @sbxchicago flexible, two-column list.

Reply:
Once we resolve that, we can merge your PR into this one and this into master.

Reply:
FYI...I've resolved this issue in 83250d9e6c658c0a8f762e8ba3b4915d20dedd29

Reply:
@markcmiller86, looks like it is fixed now looking at:

* https://preview.bssw.io/blog_posts/celebrating-apollo-s-50th-anniversary-users-stories-from-space

which shows:

![image](https://user-images.githubusercontent.com/1959736/128407891-2550a23b-f259-4923-b79c-468434d2b34e.png)